internal/controllers/clustertemplate_controller.go

```diff
@@ -96,31 +121,177 @@ func (r *ClusterTemplateReconciler) Reconcile(ctx context.Context, req ctrl.Requ
-	ctx context.Context) error {
+// validateClusterTemplateCR validates the ClusterTemplate CR and sets the Validated status condition.
+// It returns true if valid, false otherwise.
+func (t *clusterTemplateReconcilerTask) validateClusterTemplateCR(ctx context.Context) (bool, error) {
+	var validationErrs []string
 
-	t.object.Status.ClusterTemplateValidation.ClusterTemplateIsValid = true
-	t.object.Status.ClusterTemplateValidation.ClusterTemplateError = ""
+	// TODO: validition for hw template configmap
+	// TODO: validition for policy template configmap
 
-	return utils.UpdateK8sCRStatus(ctx, t.client, t.object)
+	// Validate the ClusterInstance defaults configmap
+	validationErr, err := validateConfigmapReference(
+		ctx, t.client,
+		t.object.Spec.Templates.ClusterInstanceDefaults,
+		t.object.Namespace,
+		utils.ClusterInstanceTemplateDefaultsConfigmapSuffix)
+	if err != nil {
+		return false, fmt.Errorf("failed to validate the ConfigMap %s for ClusterInstance defaults: %w",
+			t.object.Spec.Templates.ClusterInstanceDefaults, err)
+	}
+	if validationErr != "" {
+		validationErrs = append(validationErrs, validationErr)
+	}
+
+	// Validate unique ConfigMap references - ClusterInstanceDefaults
+	validationErr, err = t.validateUniqueConfigmapReferences(ctx)
+	if err != nil {
+		return false, fmt.Errorf("failed to check unique ConfigMap referencese : %w", err)
+	}
+	if validationErr != "" {
+		validationErrs = append(validationErrs, validationErr)
+	}
+
+	validationErrsMsg := strings.Join(validationErrs, ";")
+	if validationErrsMsg != "" {
+		t.logger.ErrorContext(ctx, fmt.Sprintf(
+			"Failed to validate for ClusterTemplate %s: %s", t.object.Name, validationErrsMsg))
+	} else {
+		t.logger.InfoContext(ctx, fmt.Sprintf(
+			"Validation passing for ClusterTemplate %s", t.object.Name))
+	}
+
+	t.setStatusConditionValidated(validationErrsMsg)
+	return validationErrsMsg == "", nil
+}
+
+// validateConfigmapReference validates a given configmap reference within the ClusterTemplate
+func validateConfigmapReference(
+	ctx context.Context, c client.Client, name, namespace, expectedKey string) (string, error) {
+
+	var validationErr string
+
+	existingConfigmap := &corev1.ConfigMap{}
+	cmExists, err := utils.DoesK8SResourceExist(
+		ctx, c, name, namespace, existingConfigmap)
+	if err != nil {
+		return validationErr, err
+	}
+
+	if !cmExists {
+		// Check if the configmap is missing
+		validationErr = fmt.Sprintf(
+			"The referenced ConfigMap %s is not found in the namespace %s", name, namespace)
+		return validationErr, nil
+	} else {
+		// Check if the expected key is present in the configmap data
+		defaults, exists := existingConfigmap.Data[utils.ClusterInstanceTemplateDefaultsConfigmapSuffix]
+		if !exists {
+			validationErr = fmt.Sprintf(
+				"the expected key %s does not exist in the ConfigMap %s data", expectedKey, name)
+			return validationErr, nil
+		}
+
+		// Check if the data value is either valid JSON or YAML
```

Comment:
Just wondering why we would still allow JSON if we are recommending YAML format?

Reply:
No parituclar reason. It supported both json and yaml string. I just didn't change it. Sure, we could reject JSON string and only allow YAML string in the configmap data value.

Reply:
I changed to use `yaml.Unmarshal` to parse the string, but I just found that yaml parser in Go can handle json strings as well that means any valid Json string can also be parsed as valid yaml string by `yaml.Unmarshal`. It's a superset of Json.  We should just simply remove the JSON word from the comment/error message for avoid misleading.